arch/x86/core/ia32/fatal.c

```diff
@@ -171,7 +171,14 @@ static __used void df_handler_bottom(void)
 
 	LOG_ERR("Double Fault");
 #ifdef CONFIG_THREAD_STACK_INFO
-	if (z_x86_check_stack_bounds(_df_esf.esp, 0, _df_esf.cs)) {
+	/* To comply with MISRA 13.2 rule necessary to exclude code that depends
+	 * on the order of evaluation of function arguments.
+	 * Create 2 variables to store volatile data from the structure _df_esf
+	 */
+	uint32_t df_esf_esp = _df_esf.esp;
+	uint32_t df_esf_cs = _df_esf.cs;
```

Comment:
this needs to be uint16_t

Reply:
As I understand only `df_esf_cs` should be `uint16_t`, because function's third argument is `uint16_t`
`bool z_x86_check_stack_bounds(uintptr_t addr, size_t size, uint16_t cs)`

Reply:
yes